src/main/java/gyro/aws/ec2/PeeringConnectionResource.java

```diff
@@ -301,16 +301,73 @@ private VpcPeeringConnection getVpcPeeringConnection(Ec2Client client) {
-                getAllowDnsResolutionFromRemoteVpc())
-                .allowEgressFromLocalClassicLinkToRemoteVpc(getAllowEgressFromLocalClassicLinkToRemoteVpc())
-                .allowEgressFromLocalVpcToRemoteClassicLink(getAllowEgressFromLocalVpcToRemoteClassicLink()))
-        );
+            .requesterPeeringConnectionOptions(req.build()));
+    }
+
+    private void modifyAccepterPeeringConnectionSettings(Ec2Client client) {
+        PeeringConnectionOptionsRequest.Builder acp = PeeringConnectionOptionsRequest.builder();
+
+        if (getPeerAllowDnsResolutionFromRemoteVpc()) {
+            acp.allowDnsResolutionFromRemoteVpc(getPeerAllowDnsResolutionFromRemoteVpc());
+        }
+        if (getPeerAllowEgressFromLocalClassicLinkToRemoteVpc()) {
+            acp.allowEgressFromLocalClassicLinkToRemoteVpc(getPeerAllowEgressFromLocalClassicLinkToRemoteVpc());
+        }
+        if (getPeerAllowEgressFromLocalVpcToRemoteClassicLink()) {
+            acp.allowEgressFromLocalVpcToRemoteClassicLink(getPeerAllowEgressFromLocalVpcToRemoteClassicLink());
+        }
+
+        client.modifyVpcPeeringConnectionOptions(r -> r.vpcPeeringConnectionId(getId())
+            .accepterPeeringConnectionOptions(acp.build()));
+    }
+
+    private void modifyPeeringConnectionSettings(Ec2Client client) {
+        if (!getVpc().getRegion().equals(getPeerVpc().getRegion())) {
+            try (Ec2Client accepterClient = createClient(Ec2Client.class, getPeerVpc().getRegion(), "")) {
+                accepterClient.acceptVpcPeeringConnection(
+                    r -> r.vpcPeeringConnectionId(getId()).overrideConfiguration()
+                );
+                waitForStatus(client, "active");
+                modifyRequesterPeeringConnectionSettings(client);
+                modifyAccepterPeeringConnectionSettings(accepterClient);
+            }
+        } else {
+            client.acceptVpcPeeringConnection(r -> r.vpcPeeringConnectionId(getId()));
+            waitForStatus(client, "active");
+            modifyRequesterPeeringConnectionSettings(client);
+            modifyAccepterPeeringConnectionSettings(client);
+        }
+    }
+
+    private void waitForStatus(Ec2Client client, String status) {
+        Wait.atMost(2, TimeUnit.MINUTES)
+            .checkEvery(5, TimeUnit.SECONDS)
+            .prompt(false)
+            .resourceOverrides(this, TimeoutSettings.Action.CREATE)
+            .until(() -> {
+                DescribeVpcPeeringConnectionsResponse vpcPeeringConnectionsResponse = client.describeVpcPeeringConnections(
+                    r -> r.vpcPeeringConnectionIds(getId())
+                );
+                if (!vpcPeeringConnectionsResponse.hasVpcPeeringConnections()) {
+                    return false;
+                } else {
+                    return vpcPeeringConnectionsResponse.vpcPeeringConnections().get(0).status().code().toString().equals(status);
+                }
+            });
```

Comment:
Use the given `VpcPeeringConnectionStateReasonCode` enum
Also, make sure that the `vpcPeeringConnectionsResponse.vpcPeeringConnections()` is not empty before the `get(0)` call for general null safety.

```suggestion
    private void waitForStatus(Ec2Client client, VpcPeeringConnectionStateReasonCode status) {
        Wait.atMost(2, TimeUnit.MINUTES)
            .checkEvery(5, TimeUnit.SECONDS)
            .prompt(false)
            .resourceOverrides(this, TimeoutSettings.Action.CREATE)
            .until(() -> {
                DescribeVpcPeeringConnectionsResponse vpcPeeringConnectionsResponse =
                    client.describeVpcPeeringConnections(
                        r -> r.vpcPeeringConnectionIds(getId())
                    );
                if (!vpcPeeringConnectionsResponse.hasVpcPeeringConnections() &&
                    !vpcPeeringConnectionsResponse.vpcPeeringConnections().isEmpty()) {
                    return false;
                } else {
                    return status.equals(vpcPeeringConnectionsResponse.vpcPeeringConnections().get(0).status().code());
                }
            });
    }
```